modules/P2PDominantSpeakerDetection.js

```diff
@@ -0,0 +1,106 @@
+import * as JitsiConferenceEvents from '../JitsiConferenceEvents';
+
+/**
+ * The value which we use to say, every sound over this threshold
+ * is talking on the mic.
+ * @type {number}
+ */
+const SPEECH_DETECT_THRESHOLD = 0.6;
+
+/**
+ * The <tt>P2PDominantSpeakerDetection</tt> is activated only when p2p is
+ * currently used.
+ * Listens for changes in the audio level changes of the local p2p audio track
+ * or remote p2p one and fires dominant speaker events to be able to use
+ * features depending on those events (speaker stats), to make them work without
+ * the video bridge.
+ */
+export default class P2PDominantSpeakerDetection {
+    /**
+     * Creates P2PDominantSpeakerDetection
+     * @param conference the JitsiConference instance that created us.
+     * @constructor
+     */
+    constructor(conference) {
+        this.conference = conference;
+
+        this._audioLevel = this._audioLevel.bind(this);
+        this._onP2PChanged = this._onP2PChanged.bind(this);
+        this._trackAdded = this._trackAdded.bind(this);
+
+        conference.addEventListener(
+            JitsiConferenceEvents.P2P_STATUS, this._onP2PChanged);
+    }
+
+    /**
+     * Notified when p2p state changes for current conference.
+     *
+     * @param {JitsiConference} conference - the current conference
+     * @param {bool} isP2P - are we going in p2p mode or exiting that mode
+     * @private
+     */
+    _onP2PChanged(conference, isP2P) {
+        if (isP2P) {
+            conference.addEventListener(
+                JitsiConferenceEvents.TRACK_ADDED,
+                this._trackAdded);
+            conference.statistics.addAudioLevelListener(this._audioLevel);
```

Comment:
If find storing remote audio track's SSRC  not the cleanest way of doing it. Maybe it would be better if this module would listen to the top level audio level event which will filter out the events coming from different TraceablePeerConnections:
JitsiConferenceEvents.TRACK_AUDIO_LEVEL_CHANGED
There you will get the user's ID and the actual audio level. You can compare user's Id to figure out if it's the local or remote audio level. Also disable the module or ignore event when conference is not in P2P mode (JitsiConference.isP2PActive())